src/deepsparse/server_v2/main.py

```diff
@@ -0,0 +1,262 @@
+    @app.get("/healthcheck", tags=["general"], response_model=bool)
+    @app.get("/status", tags=["general"], response_model=bool)
+    def _health():
+        return True
+
+    app._deepsparse_pool = ThreadPoolExecutor(max_workers=context.num_streams)
+
+    add_invocations_endpoint = len(server_config.endpoints) == 1
+
+    for endpoint_config in server_config.endpoints:
+        pipeline_config = _endpoint_config_to_pipeline_config(endpoint_config)
+        pipeline_config.kwargs["executor"] = app._deepsparse_pool
+
+        _LOGGER.info(f"Initializing pipeline for '{endpoint_config.name}'")
+        pipeline = Pipeline.from_config(pipeline_config, context)
+
+        _LOGGER.info(f"Adding endpoints for '{endpoint_config.name}'")
+        _add_pipeline_endpoint(app, endpoint_config, pipeline, add_invocations_endpoint)
+
+    _LOGGER.info(f"Added endpoints: {[route.path for route in app.routes]}")
+
+    return app
+
+
+def _add_pipeline_endpoint(
+    app: FastAPI,
+    endpoint_config: EndpointConfig,
+    pipeline: Pipeline,
+    add_invocations_endpoint: bool = False,
+):
+    input_schema = pipeline.input_schema
+    output_schema = pipeline.output_schema
+
+    pool: ThreadPoolExecutor = app._deepsparse_pool
+
+    @app.post(endpoint_config.endpoint, tags=["predict"], response_model=output_schema)
+    async def _predict_func(request: pipeline.input_schema):
+        return await get_running_loop().run_in_executor(pool, pipeline, request)
+
+    _LOGGER.info(f"Added '{endpoint_config.endpoint}' endpoint")
+
+    if hasattr(input_schema, "from_files"):
+        file_endpoint = endpoint_config.endpoint + "/files"
+
+        @app.post(file_endpoint, tags=["predict"], response_model=output_schema)
+        async def _predict_from_files_func(request: List[UploadFile]):
+            request = pipeline.input_schema.from_files(
+                (file.file for file in request), from_server=True
+            )
+            return await get_running_loop().run_in_executor(pool, pipeline, request)
+
+        _LOGGER.info(f"Added '{file_endpoint}' endpoint")
+
+    if add_invocations_endpoint and hasattr(input_schema, "from_files"):
+
+        @app.post("/invocations", response_model=output_schema)
+        async def _invocations(request: List[UploadFile]):
+            return RedirectResponse(file_endpoint)
+
+        _LOGGER.info("Added '/invocations' endpoint")
+
+
+def _get_scheduler(server_config: ServerConfig) -> Scheduler:
+    return (
+        Scheduler.multi_stream
+        if server_config.num_concurrent_batches > 1
+        else Scheduler.single_stream
+    )
+
+
+def _endpoint_config_to_pipeline_config(model: EndpointConfig) -> PipelineConfig:
+    if model.batch_size == 1 and model.accept_multiples_of_batch_size:
+        # dynamic batch
+        batch_size = None
+    elif model.batch_size > 1 and model.accept_multiples_of_batch_size:
+        # should still be dynamic batch
+        raise NotImplementedError
```

Comment:
This can be handled in another PR for commercial server. In theory we should be able to split up input that is divisible by batch size